R/convertCase.R

```diff
@@ -0,0 +1,66 @@
+####
+#############################################################################
+#' Modify upper and lower case for strings.
+#'
+#' Convert a character vector, all character variables in a \code{data.frame} or selected variables in a \code{GADSdat} to
+#' upper, lower or first letter upper and everything else lower case.
+#'
+#'@param x A character vector, \code{data.frame}, or \code{GADSdat}.
+#'@param case What case should the strings be converted to?
+#'@param ...	further arguments passed to or from other methods.
+#'@param vars What variables in the GADSdat should the conversion be applied to?
+#'
+#'@return Returns the converted object.
+#'
+#'@examples
+#' # for character
+#' convertCase(c("Hi", "HEllo", "greaT"), case = "upperFirst")
+#'
+#' # for GADSdat
+#' input_g <- import_DF(data.frame(v1 = 1:3, v2 = c("Hi", "HEllo", "greaT"),
+#'                           stringsAsFactors = FALSE))
+#' convertCase(input_g, case = "upperFirst", vars = "v2")
+#'
+#'
+#'@export
+convertCase <- function(x, case = c("lower", "upper", "upperFirst"), ...){
+  UseMethod("convertCase")
+}
+
+#'@export
+convertCase.character <- function(x, case = c("lower", "upper", "upperFirst"), ...){
+  case <- match.arg(case)
+  if(identical(case, "lower")) return(tolower(x))
+  if(identical(case, "upper")) return(toupper(x))
+  if(identical(case, "upperFirst")) {
+    x2 <- tolower(x)
+    substr(x2, 1, 1) <- toupper(substr(x2, 1, 1))
+    return(x2)
+  }
+}
+
+#'@export
+convertCase.data.frame <- function(x, case = c("lower", "upper", "upperFirst"), ...){
+  # only transform character variables
+  which_character <- which(sapply(x, is.character))
+
+  for(i in which_character) {
+    x[[i]] <- convertCase(x[[i]])
+  }
```

Comment:
Prefer functionals as well, but in my experience they complicate things if the goal is to modify an existing object. Any suggestions for efficient functional usage for this application?

Reply:
```r
df <- data.frame("a" = c("A", "a"), 
                 "b" = c(1,2),
                 "c" = factor(c("a", "b"))
)

which_character <- which(colnames(df) == "a")
df[, which_character] <- as.data.frame(apply(df[ , which_character, FALSE], 2, function(x){gsub("A", "a", x)}))
```
This would be one possibility, however it turned out to be more complicated as i anticipated, so i agree, keep the for loop! Also goes with the recommendation of Wickham (http://adv-r.had.co.nz/Functionals.html), to use for loops when modifying an object.

Reply:
Ah, yes, I was looking for this section but failed to find it!